marketplacetests/payment/app.py

```diff
@@ -56,22 +54,36 @@ def pin_heading(self):
         return self.marionette.find_element(*self._pin_heading_locator).text
 
     def create_pin(self, pin):
-        self.wait_for_element_displayed(*self._pin_container_locator)
-        Wait(marionette=self.marionette).until(lambda m: 'Create' in self.pin_heading)
-        self.marionette.find_element(*self._pin_container_locator).send_keys(pin)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._pin_container_locator))
+        Wait(self.marionette).until(
+            expected.element_displayed(element))
+        Wait(self.marionette).until(lambda m: 'Create' in self.pin_heading)
+        self.keyboard.send(pin)
+        self.switch_to_payment_frame()
         self.tap_pin_continue()
 
-        # Workaround click because Marionette makes the keyboard disappear
-        self.marionette.find_element(*self._pin_container_locator).click()
+        self.confirm_pin(pin)
 
-        Wait(marionette=self.marionette).until(lambda m: 'Confirm' in self.pin_heading)
-        self.marionette.find_element(*self._pin_container_locator).send_keys(pin)
+    def confirm_pin(self, pin):
+        Wait(self.marionette).until(lambda m: 'Confirm' in self.pin_heading)
+        element = Wait(self.marionette).until(
+            expected.element_present(*self._pin_container_locator))
+        Wait(self.marionette).until(
+            expected.element_displayed(element))
+        element.click()
```

Comment:
Nit: Add a comment here that the click is necessary to show the keyboard. A tap sadly does not work, and for some reason the keyboard does not show automatically as it does when it's being driven manually.